model_analyzer/config/generate/neighborhood.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from itertools import product
+from copy import deepcopy
+
+from model_analyzer.config.generate.coordinate import Coordinate
+
+
+class Neighborhood:
+    """
+    Defines and operates on a set of coordinates within a radius around
+    a 'home' coordinate
+    """
+
+    def __init__(self, search_config, coordinate_data, home_coordinate, radius):
+        """
+        Parameters
+        ----------
+        search_config: 
+            SearchConfig object
+        coordinate_data: 
+            CoordinateData object
+        home_coordinate: 
+            Coordinate object to center the neighborhood around
+        radius: Int 
+            How large of a range around the home_coordinate to define the neighborhood
+        """
+        self._search_config = search_config
+        self._coordinate_data = coordinate_data
+        self._home_coordinate = home_coordinate
+        self._radius = radius
+
+        self._neighborhood = self._create_neighborhood()
+
+    @classmethod
+    def calc_distance(cls, coordinate1, coordinate2):
+        """ 
+        Return the euclidean distance between two coordinates
+        """
+
+        distance = 0
+        for i, v in enumerate(coordinate1):
+            diff = coordinate1[i] - coordinate2[i]
+            distance += math.pow(diff, 2)
+        distance = math.sqrt(distance)
+        return distance
+
+    def get_num_initialized_points(self):
+        """ 
+        Returns the number of coordinates in the neighborhood that have a throughput
+        associated with it
+        """
+        num_initialized = 0
```

Comment:
This can be done easily with list comprehension: 
 return sum([self._coordinate_data.get_throughput(coordinate) for coordinate in self.neighborhood])

Reply:
does sum handle the 'None ' case